querybook/webapp/components/AIAssistant/AICommandBar.tsx

```diff
@@ -81,13 +63,25 @@ export const AICommandBar: React.FC<IQueryCellCommandBarProps> = forwardRef(
         );
         const [showConfirm, setShowConfirm] = useState(false);
 
-        const finalTablesToUse = useMemo(() => {
-            if (mentionedTables.length > 0) {
-                return mentionedTables;
-            } else {
-                return uniq([...tablesInQuery, ...tables]);
-            }
-        }, [tablesInQuery, mentionedTables, tables]);
+        /**
+         * There are three types of tables:
+         * 1. Tables used in the query.
+         * 2. Tables mentioned in the command.
+         * 3. Tables selected in the table selector.
+         *
+         * Regarding which tables to use:
+         * - If there are mentioned tables in the command, use those.
+         * - If there are no mentioned tables, use the tables in the query (type 1) merged with the selected tables (type 3).
+         * - However, selected tables will override the tables in the query. If a table is deselected, it will be removed from the list, including the tables in the query.
```

Comment:
even if it is deselected from tables, isn't it still in tablesInQuery

Reply:
yes, it is still in `tablesInQuery`, but will not be in the `tables` state, which is connected with the multi table selector. so the table selector can add/remove any table, including `tablesInQuery`. 

Table in the query will only be merged into `tables` when `tablesInQuery` changes.

Reply:
oh I got it, this is to reset tables with tablesInQuery whenever tablesInQueryChanges. 👍